### PR TITLE
feat: always resolve content owner

### LINKED ISSUE #165: ContentItem.owner
Similar to #164 but less complicated because `GET v1/content?include=owner` always includes the owner.

If `owner` is not in the item's data, the owner `@property` should `GET v1/users/{owner_guid}`, insert the response into the data, and return it.

### PR DESCRIPTION
**Pull request by tdstein**
Resolves #165

### CODE REVIEW

File: tests/posit/connect/__api__/v1/content/f2f37341-e21d-3d80-c698-a935ad614066.json [hunk 1]
```diff
@@ -37,7 +37,13 @@
     "default_py_environment_management": null,
     "run_as": null,
     "run_as_current_user": false,
-    "owner_guid": "87c12c08-11cd-4de1-8da3-12a7579c4998",
+    "owner_guid": "20a79ce3-6e87-4522-9faf-be24228800a4",
+    "owner": {
```
Comment (nealrichardson) [inline tests/posit/connect/__api__/v1/content/f2f37341-e21d-3d80-c698-a935ad614066.json:41]: I would back this change out. The API wouldn't return `owner` in this request.

File: src/posit/connect/content.py [hunk 2]
```diff
@@ -146,6 +147,14 @@ def bundles(self) -> Bundles:
     def permissions(self) -> Permissions:
         return Permissions(self.config, self.session, self.guid)
 
+    @property
+    def owner(self) -> ContentItemOwner:
+        if "owner" in self:
+            return self["owner"]
+        user = Users(self.config, self.session).get(self.owner_guid)
+        owner = ContentItemOwner(self.config, self.session, **user)
+        return owner
```
Comment (nealrichardson) [inline src/posit/connect/content.py:156]: How about this? We should cache the owner User.

```suggestion
        if "owner" not in self:
            # It is possible to get a content item that does not contain owner.
            # "owner" is an optional additional request param.
            # If it's not included, we can retrieve the information by `owner_guid`
            self["owner"] = Users(self.config, self.session).get(self.owner_guid)
        return ContentItemOwner(self.config, self.session, **self["owner"])
```

Comment (tdstein) [reply src/posit/connect/content.py:156]: Nice. 👏🏻

File: tests/posit/connect/test_content.py [hunk 3]
```diff
@@ -165,6 +187,30 @@ def test_tags(self):
...
+            f"https://connect.example/__api__/v1/users/20a79ce3-6e87-4522-9faf-be24228800a4",
+            json=load_mock(
+                f"v1/users/20a79ce3-6e87-4522-9faf-be24228800a4.json"
+            ),
+        )
+
+        c = Client("12345", "https://connect.example")
+        item = c.content.get("f2f37341-e21d-3d80-c698-a935ad614066")
+        owner = item.owner
+        assert owner.guid == "20a79ce3-6e87-4522-9faf-be24228800a4"
```
Comment (nealrichardson) [inline tests/posit/connect/test_content.py:211]: Other things to assert: 

* owner is ContentItemOwner (if we care about that)
* The user GET on L201 was called once
* Do item.owner again and confirm that L201 has still only been called once.

**Review by nealrichardson — APPROVED** [review]
One last suggestion on the test but otherwise LGTM!

**Pull request merged**